controllers/datadogmonitor/controller.go

```diff
@@ -0,0 +1,325 @@
+		if result, err = r.checkRequiredTags(logger, instance); err != nil || result.Requeue {
+			return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+		}
+
+		err = r.create(logger, instance, newStatus, now)
+		if err != nil {
+			logger.Error(err, "Error creating monitor")
+		}
+		newStatus.CurrentHash = instanceSpecHash
+	} else {
+		// Check if instance needs to be updated
+		if instanceSpecHash != statusSpecHash {
+			// Make sure required tags are present
+			if result, err = r.checkRequiredTags(logger, instance); err != nil || result.Requeue {
+				return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+			}
+			// Update action
+			err = r.update(logger, instance, newStatus, now)
+			if err != nil {
+				logger.Error(err, "Error updating monitor")
+			}
+			newStatus.CurrentHash = instanceSpecHash
+		} else {
+			// Spec has not changed, just check if monitor state has changed (alert, warn, OK, etc.)
+			err = r.get(logger, instance, newStatus)
+			if err != nil {
+				logger.Error(err, "Error getting monitor")
+			}
+		}
+	}
+
+	// Requeue
+	if !result.Requeue && result.RequeueAfter == 0 {
+		result.RequeueAfter = defaultRequeuePeriod
+	}
+
+	// Update the status via the Kubernetes API
+	return r.updateStatusIfNeeded(logger, instance, now, newStatus, err, result)
+}
+
+func (r *Reconciler) create(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, status *datadoghqv1alpha1.DatadogMonitorStatus, now metav1.Time) error {
+	logger.Info(fmt.Sprint(datadogMonitor.Spec.Query))
+
+	logger.Info(fmt.Sprint(datadogMonitor.Status.ID))
+	// Validate monitor in Datadog
+	err := validateMonitor(r.datadogAuth, r.datadogClient, datadogMonitor)
+	if err != nil {
+		return err
+	}
+
+	// Create monitor in Datadog
+	m, err := createMonitor(r.datadogAuth, r.datadogClient, datadogMonitor)
+	if err != nil {
+		return err
+	}
+	event := buildEventInfo(datadogMonitor.Name, datadogMonitor.Namespace, datadog.CreationEvent)
+	r.recordEvent(datadogMonitor, event)
+
+	// As this is a new monitor, add static information to status
+	status.ID = int(m.GetId())
+	creator := m.GetCreator()
+	status.Creator = creator.GetEmail()
+	createdTime := metav1.NewTime(m.GetCreated())
+	status.Created = &createdTime
+	status.Primary = true
+
+	// Set Created Condition
+	condition.UpdateDatadogMonitorConditions(status, now, datadoghqv1alpha1.DatadogMonitorConditionTypeCreated, corev1.ConditionTrue, "DatadogMonitor Created")
+	logger.Info("Created a new DatadogMonitor", "datadogMonitor.Namespace", datadogMonitor.Namespace, "datadogMonitor.Name", datadogMonitor.Name, "m.ID", m.GetId())
+
+	return nil
+}
+
+func (r *Reconciler) update(logger logr.Logger, datadogMonitor *datadoghqv1alpha1.DatadogMonitor, status *datadoghqv1alpha1.DatadogMonitorStatus, now metav1.Time) error {
+	// Validate monitor in Datadog
+	err := validateMonitor(r.datadogAuth, r.datadogClient, datadogMonitor)
+	if err != nil {
+		return err
```

Comment:
it can be good to enrich the error maybe with more context

Reply:
thanks; i added a monitor ID to the error log where `r.update` is called, but let me know if you think additional info should be added (or moved here)